Add note about base_path and ECE to the snapshot repository docs (#83526)
elastic/elasticsearch: docs/reference/snapshot-restore/repository-azure.asciidoc

    Specifies the path within container to repository data. Defaults to empty
    (root directory).
+

> NOTE: {ECE} generates the `base_path` for each deployment so that multiple
> deployments may share the same bucket. Do not set `base_path` when
> configuring a snapshot repository for {ECE}.

Reviewer: I'd flip the order of the sentences here so the imperative comes first.
```suggestion
NOTE: Don't set `base_path` when configuring a snapshot repository for {ECE}.
{ECE} automatically generates the `base_path` for each deployment so that
multiple deployments may share the same bucket.
```
